packages/twenty-front/src/modules/favorites/components/CurrentWorkspaceMemberFavorites.tsx

```diff
@@ -1,127 +1,229 @@
-import { useNavigationSection } from '@/ui/navigation/navigation-drawer/hooks/useNavigationSection';
-
-import { currentWorkspaceMemberState } from '@/auth/states/currentWorkspaceMemberState';
-import { NavigationDrawerAnimatedCollapseWrapper } from '@/ui/navigation/navigation-drawer/components/NavigationDrawerAnimatedCollapseWrapper';
 import { NavigationDrawerItemsCollapsedContainer } from '@/ui/navigation/navigation-drawer/components/NavigationDrawerItemsCollapsedContainer';
-import { ScrollWrapper } from '@/ui/utilities/scroll/components/ScrollWrapper';
-import { useFavorites } from '../hooks/useFavorites';
-
-const StyledContainer = styled(NavigationDrawerSection)`
-  width: 100%;
-`;
-
-const StyledAvatar = styled(Avatar)`
-  :hover {
-    cursor: grab;
-  }
-`;
-
-const StyledNavigationDrawerItem = styled(NavigationDrawerItem)`
-  :active {
-    cursor: grabbing;
-
-    .fav-avatar:hover {
-      cursor: grabbing;
-    }
-  }
-`;
-
-export const CurrentWorkspaceMemberFavorites = () => {
-  const currentWorkspaceMember = useRecoilValue(currentWorkspaceMemberState);
-
-  const { favorites, handleReorderFavorite } = useFavorites();
-  const loading = useIsPrefetchLoading();
-  const { toggleNavigationSection, isNavigationSectionOpenState } =
-    useNavigationSection('Favorites');
-  const isNavigationSectionOpen = useRecoilValue(isNavigationSectionOpenState);
-
-  if (loading && isDefined(currentWorkspaceMember)) {
-    return <FavoritesSkeletonLoader />;
-  }
+import { NavigationDrawerSubItem } from '@/ui/navigation/navigation-drawer/components/NavigationDrawerSubItem';
+import { getNavigationSubItemState } from '@/ui/navigation/navigation-drawer/utils/getNavigationSubItemState';
+import { useTheme } from '@emotion/react';
+import { useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import {
+  IconDotsVertical,
+  IconFolder,
+  IconHeartOff,
+  IconPencil,
+  IconTrash,
+  MenuItem,
+} from 'twenty-ui';
+
+type CurrentWorkspaceMemberFavoritesProps = {
+  folder: {
+    folderId: string;
+    folderName: string;
+    favorites: ProcessedFavorite[];
+  };
+  isGroup: boolean;
+  isOpen: boolean;
+  onToggle: (folderId: string) => void;
+};
 
-  const currentWorkspaceMemberFavorites = favorites.filter(
-    (favorite) => favorite.workspaceMemberId === currentWorkspaceMember?.id,
+export const CurrentWorkspaceMemberFavorites = ({
+  folder,
+  isGroup,
+  isOpen,
+  onToggle,
+}: CurrentWorkspaceMemberFavoritesProps) => {
+  const currentPath = useLocation().pathname;
+  const currentPathView = useLocation().pathname + useLocation().search;
+
+  const theme = useTheme();
+  const [isRenaming, setIsRenaming] = useState(false);
+  const [folderName, setFolderName] = useState(folder.folderName);
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+
+  const { renameFolder, deleteFolder } = useFavoriteFolders();
```

Comment:
I would introduce one hook by function: useRenameFavoriteFolder, useDeleteFavoriteFolder, to avoid:
- react computation: this will load all hook code for nothing
- undesired re-renders (here are performing synchronous actions, there is no need to listen on anything). If we need to listen to some states, we can use a recoilCallback